pkg/controller/operators/olm/operator.go

```diff
@@ -241,6 +279,180 @@ func (a *Operator) syncRBAC(obj interface{}) (syncError error) {
+	}
+
+	nsMap := map[string]struct{}{}
+	for _, v := range clusterNamespaces {
+		nsMap[v.Name] = struct{}{}
+	}
+	for _, v := range statusNamespaces {
+		if _, ok := nsMap[v.Name]; !ok {
+			return true
+		}
+	}
+	return false
+}
+
+func (a *Operator) updateDeploymentAnnotation(op *v1alpha2.OperatorGroup) (error, []corev1.Namespace) {
+	// NOTE: if a CSV modification is required in the future, copy the original
+	// data as done in the bottom of this method first.
+
+	selector, err := metav1.LabelSelectorAsSelector(&op.Spec.Selector)
+	if err != nil {
+		return err, nil
+	}
+	operatorGroupOpts := metav1.ListOptions{LabelSelector: selector.String()}
+	namespaceList, err := a.OpClient.KubernetesInterface().CoreV1().Namespaces().List(operatorGroupOpts)
+	if err != nil {
+		return err, nil
+	}
+
+	if !namespacesChanged(namespaceList.Items, op.Status.Namespaces) {
+		// status is current with correct namespaces, so no further updates required
+		log.Debugf("No namespace changes detected, found: %v", namespaceList.Items)
+		return nil, namespaceList.Items
+	}
+	op.Status.Namespaces = namespaceList.Items
+	op.Status.LastUpdated = timeNow()
+
+	currentNamespace := op.GetNamespace()
+	csvsInNamespace := a.csvsInNamespace(currentNamespace)
+	for csvName, csv := range csvsInNamespace {
+		strategy, err := a.resolver.UnmarshalStrategy(csv.Spec.InstallStrategy)
+		if err != nil {
+			return fmt.Errorf("error unmarshaling strategy from ClusterServiceVersion '%s' with error: %s", csvName, err), namespaceList.Items
+		}
+
+		strategyDetailsDeployment, ok := strategy.(*install.StrategyDetailsDeployment)
+		if !ok {
+			return fmt.Errorf("could not assert strategy implementation as deployment for CSV %s", csvName), namespaceList.Items
+		}
+
+		managerPolicyRules := []rbacv1.PolicyRule{}
+		apiEditPolicyRules := []rbacv1.PolicyRule{}
+		apiViewPolicyRules := []rbacv1.PolicyRule{}
+		for _, owned := range csv.Spec.CustomResourceDefinitions.Owned {
+			resourceNames := []string{}
+			for _, resource := range owned.Resources {
+				resourceNames = append(resourceNames, resource.Name)
+			}
+			managerPolicyRules = append(managerPolicyRules, rbacv1.PolicyRule{Verbs: []string{"*"}, APIGroups: []string{owned.Name}, Resources: resourceNames})
+			apiEditPolicyRules = append(apiEditPolicyRules, rbacv1.PolicyRule{Verbs: []string{"create", "update", "patch", "delete"}, APIGroups: []string{owned.Name}, Resources: []string{owned.Kind}})
+			apiViewPolicyRules = append(apiViewPolicyRules, rbacv1.PolicyRule{Verbs: []string{"get", "list", "watch"}, APIGroups: []string{owned.Name}, Resources: []string{owned.Kind}})
+		}
+		for _, owned := range csv.Spec.APIServiceDefinitions.Owned {
+			resourceNames := []string{}
+			for _, resource := range owned.Resources {
+				resourceNames = append(resourceNames, resource.Name)
+			}
+			managerPolicyRules = append(managerPolicyRules, rbacv1.PolicyRule{Verbs: []string{"*"}, APIGroups: []string{owned.Name}, Resources: resourceNames})
```

Comment:
I don't think Resources belong to the APIService's API group.

Reply:
Can you clarify this with respect to the comment made here: https://github.com/operator-framework/operator-lifecycle-manager/pull/480#discussion_r221736953